### PR TITLE
Handle 201 Created response code

### PR DESCRIPTION
**Pull request by jessm12**
# Changes

Return headers on 201 Created response code, previous behaviour for 204 and other ok responses remains the same 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by AlanGreene** [comment]
@jessm12 can you add a test to cover the new case?

File: src/api/comms.test.js [hunk 1]
```diff
@@ -38,6 +38,12 @@ describe('checkStatus', () => {
     const json = jest.fn(() => data);
     expect(checkStatus({ ok: true, json })).toEqual(data);
   });
+  
+  it('returns headers on successful create', () => {
```
Comment (AlanGreene) [inline src/api/comms.test.js]: unit test failing due to missing params to `checkStatus`: https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_dashboard/153/pull-tekton-dashboard-unit-tests/1129066520287121409/?log#log

PR open to ensure build fails as expected on node.js unit test failure: https://github.com/tektoncd/dashboard/pull/159

Unit test should look something like:

```js
it('returns headers on successful create', () => {
  const status = 201;
  const headers = { fake: 'headers' };
  expect(checkStatus({ ok: true, headers, status }).toEqual(headers);
});
```

Comment (jessm12) [reply src/api/comms.test.js]: Thanks @AlanGreene I have done that update

**Comment by AlanGreene** [comment]
@jessm12 looks good now thanks, can you squash your commits and this should be good to go

**Pull request merged**